code/game/objects/items/weapons/kitchen.dm

```diff
@@ -70,6 +70,20 @@
 	desc = "It's a fork. Sure is pointy."
 	icon_state = "fork"
 
+/obj/item/kitchen/utensil/fork/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is putting the [src] into the garbage disposal!</span>")
+	var/message = "DING DING DING DING DING DING DING DING DING"
+	var/span = "reallybig"
+	sleep(2 SECONDS)
```

Comment:
```suggestion
	if(!do_mob(user, user, 2 SECONDS) || QDELETED(user))
		return SHAME
```

Reply:
Not a fan of this one, it's making the code less readable.

Reply:
What is the problem with it? Hes not rewriting the same thing in another way.

Reply:
As Henri said, this gives the progressbar and makes it cancellable, alongside a check if the mob is qdeleted. It does more than just waiting 2 seconds